src/directives/formly-field.js

```diff
@@ -87,9 +87,9 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
       if (!key || !model) {
         return
       }
+
       // If we are working with a number then $parse wont work, default back to the old way for now
-      if (angular.isNumber(key)) {
-        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
+      if (angular.isNumber(key) || (/\d/.test(key) && key.indexOf('[') === -1)) {
```

Comment:
Yeah, that's what I'm saying. That's what `/^\d/` would do. It simply would check whether the first character is a digit.

For the if statement, I would expect something like:

```javascript

if (shouldNotUseParse(key) {
  return model[key]
} // etc.

// function down below...
function shouldNotUseParse(key) {
  return angular.isNumber(key) || (/^\d/.test(key) && key.indexOf('[') === -1)
}
```